src/opensles/AudioStreamOpenSLES.h

```diff
@@ -110,7 +112,7 @@ class AudioStreamOpenSLES : public AudioStreamBuffered {
     SLObjectItf                   mObjectInterface = nullptr;
     SLAndroidSimpleBufferQueueItf mSimpleBufferQueueInterface = nullptr;
 
-    uint8_t                      *mCallbackBuffer = nullptr;
+    std::vector<uint8_t>          mCallbackBuffer;
```

Comment:
Ah, I see @philburk already said "Good idea" on a previous comment. As you were :)

Reply:
I like the idea of making it private. I'll upload a new commit.

Reply:
@dturner - I am trying to be more modern by using vector instead of raw arrays. But please weigh in if you think there is a better way.  If we are concerned about resizing, would std::array be better?

Reply:
`std::array` is for compile-time arrays (i.e. wraps stack allocated arrays). In this case we don't know the appropriate size of the array. The best option here would be `std::unique_ptr<uint8_t[]>`. Array unique pointers were made specifically for wrapping C-style arrays in situations such as these. It has the benefit of being memory managed. Since I believe we are compiling c++14, we could also use `auto mCallbackBuffer = std::make_unique<unit8_t[]>(mBytesPerCallback)`. It preserves array referencing and *cannot* be runtime resized.

Reply:
@tguilbert-google -  Does the Chrome team have a strong preference for std::vector? If not then what do you think of Atneya's suggestion of std::unique_ptr<uint8_t[]> ?

Reply:
I don't have a preference. I will update the code